src/com/google/enterprise/gsafeed/DateAdapter.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.enterprise.gsafeed;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+import javax.xml.bind.annotation.adapters.XmlAdapter;
+
+/**
+ * Converts between XML string representation of dates in RFC 822
+ * format and java.util.Date objects during marshal/unmarshal.
+ */
+class DateAdapter extends XmlAdapter<String, Date> {
+  public static final String FORMAT = "EEE, dd MMM yyyy HH:mm:ss z";
```

Comment:
Maybe s/dd/d/ since this changed some test data from "6 Nov 2007" to "06 Nov 2007"? Is the single-digit day from GSA feed files?

Reply:
The single-digit day is from the feed example in the feed developer's guide, which is the sample data I started from for the first tests. The two-digit form is in the feed I downloaded from a GSA. The two-digit form is also used in GsaFeedFileMaker.